### DISCUSSION TITLE
DuckDB and Fugue integration is released!

### DISCUSSION THREAD

**Discussion by goodwanghan** [duckdb/duckdb#3002] [Show and tell]
Hi,

We had a [discussion](https://github.com/duckdb/duckdb/discussions/2447) in October about how to have deeper integration of DuckDB on [Fugue](https://github.com/fugue-project/fugue) (a low code abstraction for expressing framework-agnostic workflows using SQL or Python). And this week, we released Fugue [0.6.5](https://github.com/fugue-project/fugue/releases/tag/0.6.5) to include this deep integration.

As I explained, the idea is to try to keep the data inside DuckDB end to end to avoid the overhead of data conversion and transfer. Actually, another huge benefit is this approach takes full advantage of the pushdown mechanism of DuckDB, so it becomes even faster.

For example in the following Fugue SQL code:

```sql
LOAD "...parquet"
SELECT a, SUM(b) AS b GROUP BY a
PRINT
```
It can finish very quickly even with GB level parquet data because only `a` and `b` will be loaded. In addition, Fugue integrates [Ibis](https://github.com/ibis-project/ibis) and DuckDB seamlessly so people who don't like SQL can still enjoy DuckDB in a pythonic way. More examples can be found in this [tutorial page](https://fugue-tutorials.readthedocs.io/tutorials/integrations/duckdb.html).

As the next step, we may create another backend combining DuckDB and [Dask](https://dask.org/). To clarify, the semantic of `groupby().apply()` is not in the scope of SQL, but a necessary operation for data processing, that is why we need Dask to fully utilize all CPUs.

I really enjoy using DuckDB. Thank you for the amazing work!

If you don't mind I add Fugue to your [projects list](https://duckdb.org/why_duckdb.html#other-projects) please point me to the file I should change, I will make a PR.

**Comment by hannes** [comment]
Thanks for sharing, happy to hear you are enjoying using DuckDB and thank you of course for working on the integration. Happy to add Fugue to the list, file is here: https://github.com/duckdb/duckdb-web/blob/master/why_duckdb.md

**Reply by goodwanghan** [reply]
Thank you! I will make a pr to update it.
